### PR TITLE
feat(manager/fleet): support updating customisations of versions

### LINKED ISSUE #18177: Rancher's Fleet version handling problem
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### If you're self-hosting Renovate, select which platform you are using.

_No response_

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

If I start to use a renovate to manage my Helm Chart versions for [fleet.rancher.io](https://fleet.rancher.io/), I'm only able to use the functionality at top level.

There is an option in fleet that you could manage helm chart version per cluster.
Here is an example fleet.yaml file:
```yaml

defaultNamespace: cattle-monitoring-system
dependsOn:
  - name: rancher-charts-fleet-rancher-charts-01-crd-monitoring
helm:
  releaseName: rancher-monitoring
  chart: rancher-monitoring
  repo: https://charts.rancher.io/
  version: "100.1.3+up19.0.3"
targetCustomizations:
- name: rke2
  clusterSelector:
    matchLabels:
      provider.cattle.io: rke2
      topology.kubernetes.io/region: budapest
      topology.kubernetes.io/zone: my-zone
  helm:
    version: "100.1.3+up19.0.3"
    valuesFiles:
      - rke2.yaml
targetCustomizations:
- name: rke2-2
  clusterSelector:
    matchLabels:
      provider.cattle.io: rke2
      topology.kubernetes.io/region: budapest
      topology.kubernetes.io/zone: my-zone-2
  helm:
    version: "100.1.2+up19.0.3"
    valuesFiles:
      - rke2-2.yaml
```
In this usecase renovate only uses 
```yaml
helm:
  version: 1.1
``` 

I trie to use it in my homelab repo:
https://github.com/marthydavid/homelab

### Relevant debug logs

<details><summary>Logs</summary>

```
        "packageFile": "fleet/rancher-charts/01-monitoring/fleet.yaml",
        "deps": [
          {
            "depType": "fleet",
            "datasource": "helm",
            "depName": "rancher-monitoring",
            "registryUrls": [
              "https://charts.rancher.io/"
            ],
            "currentValue": "100.1.3+up19.0.3",
            "depIndex": 0,
            "updates": [],
            "warnings": [],
            "versioning": "helm",
            "sourceUrl": "https://github.com/prometheus-community/helm-charts",
            "homepage": "https://github.com/prometheus-operator/kube-prometheus",
            "currentVersion": "100.1.3+up19.0.3",
            "fixedVersion": "100.1.3+up19.0.3"
          }
        ]
      },
```

</details>

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

### PR DESCRIPTION
**Pull request by secustor**
## Changes
Allows to update version overwrites of the base helm chart definition.     

The customizations name is set as `depName`.

## Context
- Closes #18177

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/fleet/extract.spec.ts [hunk 1]
```diff
@@ -61,6 +64,49 @@ kind: Fleet
...
+            datasource: 'helm',
+            depName: 'cert-manager',
+            packageName: 'cert-manager',
+            registryUrls: ['https://charts.jetstack.io'],
+            depType: 'fleet',
+          },
+          {
+            currentValue: 'v1.9.2',
+            datasource: 'helm',
+            depName: 'rke2',
```
Comment (viceice) [inline lib/modules/manager/fleet/extract.spec.ts:86]: i think this different depName will confuse users.

Comment (secustor) [reply lib/modules/manager/fleet/extract.spec.ts:86]: 🤔 My reasoning to implement it like this is, that users will for sure want separate PRs for each customization. At least by default, but maybe there is a better way to do this.

**Review by viceice — COMMENTED** [review]
see comment

File: lib/modules/manager/fleet/extract.spec.ts (hunk 1 above)
Comment (rarkins) [reply lib/modules/manager/fleet/extract.spec.ts:86]: I am ok with the proposed change

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**